### PR TITLE
Simplify example

### LINKED ISSUE #23981: improve nullability code sample
the doc sample code is currently
public static bool IsValid(Person? person)
{
    return person != null && !string.IsNullOrEmpty(person.Name);
}
but should [simplify] be
public static bool IsValid(Person? person)
{
    return !string.IsNullOrEmpty(person?.Name);
}
or even [better]
public static bool IsValid(Person? person) => !string.IsNullOrEmpty(person?.Name);

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0471f98c-f5cd-00d2-a705-396682f73bcb
* Version Independent ID: a24b862d-c462-a040-8470-9771c74c4e6f
* Content: [! (null-forgiving) operator - C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving)
* Content Source: [docs/csharp/language-reference/operators/null-forgiving.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/operators/null-forgiving.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/docs/issues/23981

### CODE REVIEW

**Comment by pkulikov** [comment]
It's a good refactoring in general. However, I'm afraid it may make understanding the containing article more difficult. See the article for the context:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving

Now, it's clear from the first look that if the `IsValid` method returns true, its argument is non-null. With the introduced change, a reader should spend more time to reason about the example. Not sure if that's necessary.

File: docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs [hunk 1]
```diff
@@ -39,9 +39,7 @@ public static void Main()
         }
 
         public static bool IsValid(Person? person)
-        {
-            return person != null && !string.IsNullOrEmpty(person.Name);
-        }
+            => !string.IsNullOrEmpty(person?.Name);
```
Comment (IEvangelist) [inline docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:42]:
```suggestion
            => person is { Name: { Length: > 0 } };

```

File: docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs [hunk 2]
```diff
@@ -60,9 +58,7 @@ public static void Main()
         }
 
         public static bool IsValid([NotNullWhen(true)] Person? person)
-        {
-            return person != null && !string.IsNullOrEmpty(person.Name);
-        }
+            => !string.IsNullOrEmpty(person?.Name);
```
Comment (IEvangelist) [inline docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:61]:
```suggestion
            => person is { Name: { Length: > 0 } };
```

File: docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs (hunk 1 above)
Comment (BillWagner) [inline docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:42]: I'm curious about this change instead:

```csharp
public static bool IsValid(Person? person) => person is {​​​​​​​ Name: {​​​​​​​ Length: > 0 }​​​​​​​ }​​​​​​​;
```

Thoughts?

Comment (pkulikov) [reply docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:42]: It's a good example of pattern matching that in the context of this article may be a bit overwhelming (like I noticed above)

Comment (pkulikov) [reply docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:42]: What about `person is not null && person.Name is not null`
It's not refactoring, though, but would fit the article

Comment (BillWagner) [reply docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs:42]: I like that version from @pkulikov It is semantically different, but does highlight the concepts for the article better than the existing code, or the other suggestions.

Comment (Youssef1313) [reply docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs]: > What about `person is not null && person.Name is not null`

Looks good. Actually, even `person is not null` would be enough for the article. But I went for your suggestion.

**Review by BillWagner — APPROVED** [review]
Thanks for updating this, and for the discussion @Youssef1313 @pkulikov @IEvangelist 

This is ready to merge.

**Pull request merged**